KnowYourEnemyMutagen/Program.cs

```diff
@@ -39,315 +29,219 @@ public static int Main(string[] args)
+                        state.PatchMod.Spells.GetOrAddAsOverride(s);
                     }
                 }
             }
-
-            // ***** PART 3 *****
+            
             // Edit the effect of silver weapons
 
             if (patchSilverPerk)
             {
                 if (state.LoadOrder.ContainsKey(ModKey.FromNameAndExtension("Skyrim Immersive Creatures.esp")))
                     Console.WriteLine("WARNING: Silver Perk is being patched, but Skyrim Immersive Creatures has been detected in your load order. Know Your Enemy's silver weapon effects will NOT work against new races added by SIC.");
 
-                FormKey silverKey = Skyrim.Perk.SilverPerk;
-                FormKey dummySilverKey = new FormKey("know_your_enemy.esp", 0x0BBE10);
-                if (state.LinkCache.TryLookup<IPerkGetter>(silverKey, out var silverPerk) && silverPerk != null)
+                var silverKey = Skyrim.Perk.SilverPerk;
+                var dummySilverKey = new FormKey("know_your_enemy.esp", 0x0BBE10);
+                if (state.LinkCache.TryLookup<IPerkGetter>(silverKey, out var silverPerk))
                 {
                     Console.WriteLine("silverPerk resolved");
-                    if (state.LinkCache.TryLookup<IPerkGetter>(dummySilverKey, out var dummySilverPerk) && dummySilverPerk != null && dummySilverPerk.Effects != null)
+                    if (state.LinkCache.TryLookup<IPerkGetter>(dummySilverKey, out var dummySilverPerk))
                     {
                         Console.WriteLine("dummySilverPerk resolved");
                         Perk kyePerk = silverPerk.DeepCopy();
                         kyePerk.Effects.Clear();
-                        foreach(APerkEffect eff in dummySilverPerk.Effects)
+                        foreach (var aPerkEffectGetter in dummySilverPerk.Effects)
                         {
+                            var eff = (APerkEffect) aPerkEffectGetter;
                             kyePerk.Effects.Add(eff);
                         }
+
                         state.PatchMod.Perks.GetOrAddAsOverride(kyePerk);
                     }
                 }
             }
-
-            // ***** PART 4 *****
+            
             // Adjust traits to accommodate CACO if present
             if (state.LoadOrder.ContainsKey(ModKey.FromNameAndExtension("Complete Alchemy & Cooking Overhaul.esp")))
             {
                 Console.WriteLine("CACO detected! Adjusting kye_ab_undead and kye_ab_ghostly spells.");
-                FormKey kye_ab_ghostly_key = new FormKey("know_your_enemy.esp", 0x060B93);
-                FormKey kye_ab_undead_key = new FormKey("know_your_enemy.esp", 0x00AA43);
-                if (state.LinkCache.TryLookup<Mutagen.Bethesda.Skyrim.ISpellGetter>(kye_ab_ghostly_key, out var kye_ab_ghostly) && kye_ab_ghostly != null)
+                var kyeAbGhostlyKey = new FormKey("know_your_enemy.esp", 0x060B93);
```

Comment:
Just wondering why you use vars so much? I like explicit types wherever possible